plugins/nordic-hid/fu-nordic-hid-cfg-channel.c

```diff
@@ -1086,12 +1086,21 @@ fu_nordic_hid_cfg_channel_direct_discovery(FuNordicHidCfgChannel *self, GError *
 	 * (other instances reject the configuration channel operations).
 	 */
 	if (!fu_nordic_hid_cfg_channel_get_board_name(self, &error_board_name)) {
-		g_set_error(error,
-			    FWUPD_ERROR,
-			    FWUPD_ERROR_NOT_SUPPORTED,
-			    "Get board name failed: %s",
-			    error_board_name->message);
-		return FALSE;
+		/* If the HID device is connected over BLE, the configuration channel operation
+		 * right after reconnection may fail with an ioctl error. Retry after a delay to
+		 * ensure that the device will be properly recognized by the fwupd tool.
+		 */
+		fu_device_sleep(device, 1000);
```

Comment:
1 second is a loooong time! I think it would be better to use `fu_device_retry_full()` with a count of 100 and a delay of 10ms and then call something like `fu_nordic_hid_cfg_channel_get_board_name_cb()`.

Reply:
Thanks for the suggestion. Changed to `fu_device_retry_full`. That would allow to fine-tune the retry parameters in the future if needed too. I also shortened the retry period (I made additional checks and verified that the device is still successfully discovered with shorter retry time).